### PR TITLE
Feat(STADTPULS-318): Make all data downloadable with Web Workers

### PR DESCRIPTION
**Pull request by vogelino**
This PRs defers downloads to a web worker to avoid blocking the main thread. It shows the downloads in the header. Each download can be canceled and clicking on a download leads to the relevant sensor page.

### CODE REVIEW

File: buildWorkers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    onRebuild(error, result) {
+      if (error) console.error('watch build failed:', error)
+      else console.log('watch build succeeded:', result)
+    },
+  } : false,
+})
+  .then(() => { console.log("Build succeeded.") })
+  .catch((e) => {
+    console.log("Error building:", e.message);
+    process.exisst(1)
```
Comment (dnsos) [inline buildWorkers.js:22]: Is `exisst` really a function? Or a typo?

File: src/components/CircularProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import colors from "../../style/colors";
+
+export interface CircularProgressBarPropType
+  extends Partial<CircularProgressbarProps> {
+  percentage?: number;
+  children?: ReactNode;
+}
+
+export const CircularProgressBar: FC<CircularProgressBarPropType> = ({
+  percentage = 66,
```
Comment (dnsos) [inline src/components/CircularProgressBar/index.tsx:17]: Why default to 66?

File: src/lib/requests/getRecordsBySensorId/index.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { supabase } from "@auth/supabase";
 import { definitions } from "@common/types/supabase";
 
+const MAX_DOWNLOADABLE_RECORDS = 1000000;
+const NEXT_PUBLIC_SUPABASE_MAX_ROWS = 5;
```
Comment (dnsos) [inline src/lib/requests/getRecordsBySensorId/index.ts:5]: Shouldn't this be defined in an env variable? And set to something other than 5?

**Review by dnsos — CHANGES_REQUESTED** [review]
Very nice work with the web workers. Also the UX feels great. Fantastic work overall!

I noticed, however, that a download of about 5000 records fails (without any error message, neither in the UI nor in the console). Test case was "All data" for [this sensor](https://stadtpuls-frontend-git-feat-stadtpu-e5a0ff-technologiestiftung1.vercel.app/sensors/2) - tested in Chrome and Firefox, macOS. The download first progresses in the UI, but when 100% are reached, no download is triggered. Very small subsets work, but it actually also happens when I try to download a subset of about 250 records. (Maybe it's somehow related to my browser settings?!)

One small visual flaw: When the download queue is in focus at the moment the progress reaches 100%, two visual artifacts remain on the screen, see screenshot:

Screenshot 2021-11-18 at 10 21 26

File: buildWorkers.js (hunk 1 above)
Comment (vogelino) [reply buildWorkers.js:22]: Typo! thx good catch!

File: src/components/CircularProgressBar/index.tsx (hunk 2 above)
Comment (vogelino) [reply src/components/CircularProgressBar/index.tsx:17]: True, I though I had changed that to 0. Will change! Thx

File: src/lib/requests/getRecordsBySensorId/index.ts (hunk 3 above)
Comment (vogelino) [reply src/lib/requests/getRecordsBySensorId/index.ts:5]: True! I'm sorry, I think I'm quite tired... Will change!

**Comment by vogelino** [comment]
> I noticed, however, that a download of about 5000 records fails (without any error message, neither in the UI nor in the console).

True! This happens in some cases where the progress reaches 100% but no CSV result is yet available. I added some check to make sure this doesn't happen anymore! Good catch!

> One small visual flaw: When the download queue is in focus at the moment the progress reaches 100%, two visual artifacts remain on the screen, see screenshot:

True! I added a check that makes sure that the caret is never displayed when the content of the dropdown is empty. Good catch as well!

**Review by dnsos — APPROVED** [review]
Very nice, thanks for all the changes! The downloads are working as expected now 💯

**Pull request merged**